### ISSUE TITLE
completionHandler called but the file is not there

### ISSUE THREAD

**Issue by pvinis** [AFNetworking/AFNetworking#3617]
on 10.9, i call `downloadTaskWithRequest:progress:destination:completionHandler:` to download a file, and i get a status code of `200` back, but when the completionHandler is called, the file is not yet there.. it happens like 80% of the time.

**Comment by eamonn-webster** [comment]
I had a "similar issue", but in my case it turned out that the file wasn't being updated because it already existed. Perhaps try observing AFURLSessionDownloadTaskDidFailToMoveFileNotification, and see if there is an error.

**Comment by hehawjq** [comment]
@pvinis I also have a same question.Do you resolve it?

**Comment by eamonn-webster** [comment]
My problem was different, the file wasn't updating. What happens is that once the file has been downloaded, it calls the destination proc to determine where to move it to. So to determine whether it was the move that was failing I observed AFURLSessionDownloadTaskDidFailToMoveFileNotification to find out why. In my case it was that the file already existed. I had started by logging the value returned from the destination proc.
